src/main/java/org/jenkinsci/plugins/parallel_test_executor/MultipleBinaryFileParameterFactory.java

```diff
@@ -0,0 +1,96 @@
+            this.parameterName = parameterName;
+            this.filePattern = filePattern;
+        }
+        public final String parameterName;
+        public final String filePattern;
+    }
+
+    private final List<ParameterBinding> parametersList;
+    private final FileBuildParameterFactory.NoFilesFoundEnum noFilesFoundAction;
+
+    @DataBoundConstructor
+    public MultipleBinaryFileParameterFactory(List<ParameterBinding> parametersList, FileBuildParameterFactory.NoFilesFoundEnum noFilesFoundAction) {
+        this.parametersList = parametersList;
+        this.noFilesFoundAction = noFilesFoundAction;
+    }
+
+    public MultipleBinaryFileParameterFactory(List<ParameterBinding> parametersList) {
+        this(parametersList, FileBuildParameterFactory.NoFilesFoundEnum.SKIP);
+    }
+
+    public FileBuildParameterFactory.NoFilesFoundEnum getNoFilesFoundAction() {
+        return noFilesFoundAction;
+    }
+
+    @Override
+    public List<AbstractBuildParameters> getParameters(AbstractBuild<?, ?> build, TaskListener listener) throws IOException, InterruptedException, AbstractBuildParameters.DontTriggerException {
+        List<AbstractBuildParameters> result = Lists.newArrayList();
+        int totalFiles = 0;
+        for (final ParameterBinding parameterBinding : parametersList) {
+            // save them into the master because FileParameterValue might need files after the slave workspace have disappeared/reused
+            FilePath target = new FilePath(build.getRootDir()).child("parameter-files");
+                int k = build.getWorkspace().copyRecursiveTo(parameterBinding.filePattern, target);
+                totalFiles += k;
+                if (k > 0) {
+                    for (final FilePath f : target.list(parameterBinding.filePattern)) {
+                        LOGGER.fine("Triggering build with " + f.getName());
+
+                        result.add(new AbstractBuildParameters() {
+                            @Override
+                            public Action getAction(AbstractBuild<?, ?> build, TaskListener listener) throws IOException, InterruptedException, DontTriggerException {
+                                assert f.getChannel() == null;    // we copied files locally. This file must be local to the master
+                                FileParameterValue fv = new FileParameterValue(parameterBinding.parameterName, new File(f.getRemote()), f.getName());
+                                return new ParametersAction(fv);
+                            }
+                        });
+                    }
+                }
+        }
+        if (totalFiles ==0) {
+            noFilesFoundAction.failCheck(listener);
+        }
+
+        return result;
+    }
+
+
+    @Extension
+    public static class DescriptorImpl extends AbstractBuildParameterFactoryDescriptor {
+        @Override
+        public String getDisplayName() {
+            return "Multiple Binary Files (not meant to be used)";
```

Comment:
So can we just delete the descriptor? Or hide it via `DescriptorVisibilityFilter` if not?